pkg/db/advisory_locks.go

```diff
@@ -59,15 +60,15 @@ func (f *AdvisoryLockFactory) NewAdvisoryLock(ctx context.Context, id string, lo
 	// obtain the advisory lock (blocking)
 	if err := lock.lock(); err != nil {
 		UpdateAdvisoryLockCountMetric(lockType, "lock error")
-		errMsg := fmt.Sprintf("error obtaining the advisory lock for id %s with (%d, %d), %v",
-			id, hash(*lock.id), hash(string(*lock.lockType)), err)
+		errMsg := fmt.Sprintf("error obtaining the advisory lock for id %s type %s, %v", id, lockType, err)
 		log.Error(errMsg)
 		// the lock transaction is already started, if error happens, we return the transaction id, so that the caller
 		// can end this transaction.
 		return *lock.uuid, fmt.Errorf(errMsg)
 	}
 
-	f.locks[fmt.Sprintf("%s-%s", id, lockType)] = lock
+	log.Info(fmt.Sprintf("Locked advisory lock id=%s type=%s - owner=%s", id, lockType, *lock.uuid))
+	f.locks[*lock.uuid] = lock
```

Comment:
Good findings 👍 
the problem is in the past, if you request lock for the same resource, the f.locks map can be overwritten by the new request, because the key (the id and lockType) is the same for the same resource. the lock cannot be released due to you cannot find it in the f.locks map.